test/org/traccar/BaseTest.java

```diff
@@ -40,7 +40,7 @@ public boolean isLatestPosition(Position position) {
             @Override
             public boolean lookupAttributeBoolean(
                     long deviceId, String attributeName, boolean defaultValue, boolean lookupConfig) {
-                return defaultValue;
+                return (deviceId == 2 && attributeName == "tk103.alternative" && lookupConfig) ? true : defaultValue;
```

Comment:
Just FYI.

During debugging testcases, i found, that Context can have different IdentityManager, depending of in what order tests are run. This is because Context is a static stuff, so it will just contain IdentityManager from last test case (it can be based on FilterHandlerTest (below) or BaseTest (here) ).

So i had to modify both for now...

Reply:
We need to find a better way. I think you should just use setter like we do for decoder configuration parameters.

Reply:
Context is static. So, even if i implement some other IdentityManager in BaseTest/FilterHandlerTest, which will have some setter for Attributes, i had to set some attribute before test and unset it after test (using try-finally), so it will not change behavior for other tests. But, as you mentioned previously you does not like try-finally. I am understanding you correctly about "use setter like we do for decoder configuration parameters"? If not - can you please share a link to code snippet with setter for decoder configuration parameters.

Anyway. If this problem is not already solved in any other xxxProtocolEncoderTest (and there are nothig related to this in GT06 protocol, which you mentioned before), and as this is not very related to T580w GPS/LBS/WIFI positioning actually. Do you agree to remove testcases for commands, which shares same name? There are still a lot of tests for other commands. And then try to solve this problem separately?